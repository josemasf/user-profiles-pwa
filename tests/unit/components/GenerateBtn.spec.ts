import {
  shallowMount
} from '@vue/test-utils'
import {
  GenerateBtn
} from '@/components'

describe('GenerateBtn.vue', () => {
  const wrapper: any = shallowMount(GenerateBtn);

  describe('Rendering', () => {
    it('Se renderiza el componente', async () => {
      expect(wrapper).toMatchSnapshot();
    });
  })
  describe('Eventos', () => {
    it('emite evento generate-profiles', () => {
      const spy = jest.spyOn(wrapper.vm, 'generateProfiles');
      wrapper.vm.generateProfiles();

      expect(wrapper.emitted('generate-profiles')).toBeTruthy();
      expect(spy).toHaveBeenCalled();
    })
  })
})