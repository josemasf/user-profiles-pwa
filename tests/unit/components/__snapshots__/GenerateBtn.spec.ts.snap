// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenerateBtn.vue Rendering Se renderiza el componente 1`] = `
<v-container-stub tag="div">
  <v-row-stub tag="div" class="text-center">
    <v-col-stub cols="12" tag="div">
      <v-btn-stub elevation="2" tag="button" activeclass="" type="button">Generate profiles</v-btn-stub>
    </v-col-stub>
  </v-row-stub>
</v-container-stub>
`;

exports[`GenerateBtn.vue Rendering should trigger changed-status 1`] = `
<v-container>
  <v-row class="text-center">
    <v-col cols="12">
      <v-btn elevation="2">Generate profiles</v-btn>
    </v-col>
  </v-row>
</v-container>
`;
